app/src/main/java/org/coepi/android/ui/extensions/rx/ObservableExtensions.kt

```diff
@@ -0,0 +1,31 @@
+package org.coepi.android.ui.extensions.rx
+
+import io.reactivex.Observable
+import io.reactivex.Observable.empty
+import io.reactivex.Observable.just
+import org.coepi.android.system.rx.VoidOperationState
+import org.coepi.android.system.rx.VoidOperationState.Failure
+import org.coepi.android.system.rx.VoidOperationState.Progress
+import org.coepi.android.system.rx.VoidOperationState.Success
+import org.coepi.android.ui.common.UINotificationData
+
+/**
+ * Maps to error notifications and optionally a success notification for the operation state.
+ */
+fun Observable<VoidOperationState>.toNotification(successMessage: String? = null): Observable<UINotificationData> =
+    flatMap {
+        when (it) {
+            is Success -> successMessage?.let {
+                just(UINotificationData.Success(successMessage))
+            }
+            is Failure -> just(UINotificationData.Failure(
+                it.t.message ?: "Unknown error"
```

Comment:
Error message should be put in strings.xml.

Reply:
This will stay like this for now, since resources can't be passed here. It requires a more refined error management that uses enums and maps them to strings near to the UI, but we don't have time now to implement this. The idea is to add improvements iteratively.